masim.c

```diff
@@ -14,6 +14,16 @@
 #include "misc.h"
 #include "masim.h"
 
+// hugepage definitions
+#define PROTECTION (PROT_READ | PROT_WRITE)
+
+/* Only ia64 requires this */
+#ifdef __ia64__
+#define FLAGS (MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB | MAP_FIXED)
+#else
+#define FLAGS (MAP_PRIVATE | MAP_ANONYMOUS | MAP_HUGETLB)
+#endif
+int hugepage = 0;
```

Comment:
I'd prefer using a clearer name. e.g., `use_hugetlb`.

I thought I made this very same comment to the previous revision, but seems I messed up something.